frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx

```diff
@@ -0,0 +1,433 @@
+          case ELEMENT_TYPES.CHECKBOX:
+            children = (
+              <React.Fragment>
+                <div className="checkbox">
+                  <label>
+                    <input
+                      type="checkbox"
+                      className="create-storage-class-form__checkbox"
+                      onChange={(event) =>
+                        handleTypeParamChange(key, event, ELEMENT_TYPES.CHECKBOX)
+                      }
+                      checked={_.get(typeParamsData, `${key}.value`, false)}
+                      id={`network-type-params-${key}-checkbox`}
+                    />
+                    {_.get(parameter, 'name', key)}
+                  </label>
+                </div>
+                <HelpBlock>{validationMsg || null}</HelpBlock>
+              </React.Fragment>
+            );
+            break;
+          case ELEMENT_TYPES.DROPDOWN:
+            children = (
+              <React.Fragment>
+                <label
+                  className={classNames('control-label', { 'co-required': parameter.required })}
+                >
+                  {_.get(parameter, 'name', key)}
+                </label>
+                <Dropdown
+                  title={parameter.hintText}
+                  items={parameter.values}
+                  dropDownClassName="dropdown--full-width"
+                  selectedKey={_.get(typeParamsData, `${key}.value`)}
+                  onChange={(event) => handleTypeParamChange(key, event, ELEMENT_TYPES.DROPDOWN)}
+                />
+                <HelpBlock>{validationMsg || null}</HelpBlock>
+              </React.Fragment>
+            );
+            break;
+          case ELEMENT_TYPES.TEXT:
+          default:
+            children = (
+              <React.Fragment>
+                <label
+                  className={classNames('control-label', {
+                    'co-required': parameter.required,
+                  })}
+                >
+                  {_.get(parameter, 'name', key)}
+                </label>
+                <FormControl
+                  type="text"
+                  bsClass="pf-c-form-control"
+                  value={_.get(typeParamsData, `${key}.value`, '')}
+                  onChange={(event) => handleTypeParamChange(key, event, ELEMENT_TYPES.TEXT)}
+                />
+                <HelpBlock>{validationMsg || null}</HelpBlock>
+              </React.Fragment>
+            );
+        }
+
+        return (
+          <FormGroup
+            key={key}
+            controlId={`network-type-parameters-${key}`}
+            validationState={_.get(typeParamsData, `${key}.validationMsg`, null) ? 'error' : null}
+          >
+            {children}
+          </FormGroup>
+        );
+      });
+
+      return (
+        <React.Fragment>
+          {dynamicContent}
+          <div>L2 is the default if nothing else is provided.</div>
+        </React.Fragment>
+      );
+    };
+
+    return getTypeElements();
```

Comment:
why returning the result of `getTypeElements()` when you can have a function body here directly' ?

Reply:
Fixed!